### PR TITLE
3_lambda++ part 2 and last

### PR DESCRIPTION
**Pull request by grosu**
@andreistefanescu, the only thing left is to make the non-deterministic tests work.  Because of that, I think it is best if you revise it.

@cos, you are welcome to take a look, too.

### CODE REVIEW

File: k-distribution/tutorial/1_k/3_lambda++/lesson_3/tests/config-det.xml [hunk 1]
```diff
@@ -8,8 +8,8 @@
            exclude="omega closed-variable-capture free-variable-capture" />
 <!-- The first program, omega, does not terminate.  The last two
      are excluded because they yield nondeterminism unless we kompile with the
-     superstrict "strict" option, which is done in config2.xml.  We do not want to kompile
-     with the superstrict "strict" option for all programs, because it is too slow
-	 (we should check if it still is too slow now).  -->
+     superstrict "strict" option, which is done in config2.xml.  We do not
```
Comment (andreistefanescu) [inline k-distribution/tutorial/1_k/3_lambda++/lesson_3/tests/config-det.xml:11]: is the indentation ok?

File: k-distribution/tutorial/1_k/3_lambda++/lesson_4/NOTES.md [hunk 2]
```diff
@@ -18,3 +18,12 @@ applied.  If you make application seqstrict(2,1) then you get 4.
 Dec 06, 2014: Looks like we should discuss the --search and --transition
 options before this lesson, and then kompile the definition with option
 --transition = computational and krun it with --search.
+
+
+- The README.md says "One is to make `+` `seqstrict` in the semantics, to
+enforce its evaluation from left-to-right.  Do it and then run the program
+above again;".  Then it continues and says "The problem is now the
+non-deterministic evaluation strategy of the function application construct".
+Add this is an exercise, asking reader to fix this non-determinism.
```
Comment (andreistefanescu) [inline k-distribution/tutorial/1_k/3_lambda++/lesson_4/NOTES.md]: Add this as an exercise?

**Comment by andreistefanescu** [comment]
lgtm

**Comment by cos** [comment]
lgtm

**Comment by grosu** [comment]
OK, done with the small issues and the tests passed.  Please merge, @andreistefanescu, when you are happy with it.

**Pull request merged**